File: TheAnvil/src/main/java/com/qualityplus/anvil/TheAnvil.java
package com.qualityplus.anvil;

import com.qualityplus.anvil.api.TheAnvilAPI;
import com.qualityplus.assistant.lib.eu.okaeri.injector.annotation.Inject;
import com.qualityplus.assistant.lib.eu.okaeri.platform.core.annotation.Scan;
import com.qualityplus.assistant.okaeri.OkaeriSilentPlugin;
import lombok.Getter;

@Scan(deep = true)
public final class TheAnvil extends OkaeriSilentPlugin {
    private static @Inject
    @Getter TheAnvilAPI api;
}

File: TheAnvil/src/main/java/com/qualityplus/anvil/TheAnvil.kt
package com.qualityplus.anvil

import com.qualityplus.anvil.api.TheAnvilAPI
import com.qualityplus.assistant.okaeri.OkaeriSilentPlugin
import eu.okaeri.injector.annotation.Inject
import eu.okaeri.platform.core.annotation.Scan
import lombok.Getter

@Scan(deep = true)
class TheAnvil() : OkaeriSilentPlugin() {
    companion object Api {
        @Inject
        lateinit var api: TheAnvilAPI

        fun getApi(): TheAnvilAPI {
            return api;
        }
    }
}